packages/realm-react/src/RealmProvider.tsx

```diff
@@ -158,14 +162,26 @@ export function createRealmProviderFromConfig(
       }
     }, [realm]);
 
+    const [progress, setProgress] = useState<number | null>(null);
+
     useEffect(() => {
       const realmRef = currentRealm.current;
       // Check if we currently have an open Realm. If we do not (i.e. it is the first
       // render, or the Realm has been closed due to a config change), then we
       // need to open a new Realm.
       const shouldInitRealm = realmRef === null;
       const initRealm = async () => {
-        const openRealm = await Realm.open(configuration.current);
+        const openRealmPromise = Realm.open(configuration.current);
+        if (configuration.current.sync?.flexible) {
+          try {
+            openRealmPromise.progress((estimate: number) => {
+              setProgress(estimate);
+            });
+          } catch (error) {
+            console.warn("Progress information with @realm/react work with realm version >=12.12.0.");
```

Comment:
Not a fan of this. Should we consider bumping up the compatibility instead?

Reply:
That would be fine by me, especially if we signal it through the `peerDependencies`.